WL- 3675 Adding newUserID for template when status is userId updation.

surya-janani opened the pull request:
When the account status is userIdUpdation then get the user from the newly added method in base user directory.

surya-janani commented:
linked with https://github.com/ox-it/wl-user/pull/5

surya-janani commented:
linked with https://github.com/ox-it/wl-kernel/pull/76

surya-janani commented:
Overloading the method CreateValidationAccount to have extra field for newEmailAddress and rebased whole thing

buckett commented on account-validator-api/src/java/org/sakaiproject/accountvalidator/logic/ValidationLogic.java:
```diff
@@ -100,8 +110,8 @@
 	 * @param toSave
 	 */
 	public void save(ValidationAccount toSave);
-	
```
Can you not make un-needed whitspace changes please as it means things don't merge as cleanly as they could.

buckett commented on account-validator-tool/src/java/org/sakaiproject/accountvalidator/tool/otp/AcountValidationLocator.java:
```diff
@@ -262,8 +262,10 @@ public SecurityAdvice isAllowed(String userId, String function, String reference
 			}
-				
-	        	UserEdit u = userDirectoryService.editUser(userId);
+				if(item.getAccountStatus().equals(ValidationAccount.ACCOUNT_STATUS_USERID_UPDATE)) {
```
Indentation looks wrong.

buckett replied:
Actually, not sure as following lines aren't the same as previous....

buckett commented on account-validator-tool/src/webapp/js/accountValidator.js:
```diff
@@ -104,7 +104,13 @@ VALIDATOR.validatePassword = function() {
+		var pw2 = VALIDATOR.get("passrow2::password2").value;
+	}
+	else {
+		pw2 = pw;
```
Should be a var on this line. as it's the declaration of `pw2`.

buckett commented on account-validator-api/src/java/org/sakaiproject/accountvalidator/logic/ValidationLogic.java:
```diff
@@ -72,7 +72,17 @@
+	 * @param accountStatus
+	 * @return
+	 */
+	public ValidationAccount createValidationAccount(String userRef,Integer accountStatus,String newUserId);
```
Why would you ever have an account status of anything other than `ACCOUNT_STATUS_USERID_UPDATE` for this call?

surya-janani replied:
because it is being called by other method CreateValidationAccount(userId,accountStatus)

buckett replied:
Although internally it might call another method, having a public API include parameters it doesn't need to makes the API harder to understand.

surya-janani commented:
I've made review changes,will rebase it once I get my keyboard shift keys working properly

buckett commented on account-validator-tool/src/webapp/js/accountValidator.js:
```diff
@@ -229,6 +235,10 @@ VALIDATOR.display = function(element, show) {
 
 // Original document ready function
 $(document).ready(function() {
+	/* Hide div with yellow background if it is empty */
```
Indentation looks wrong.

ouit0408 replied:
Fixed the indentation.

ouit0408 committed:
WL-3675 Adding a new account status for userID updation.

Adding newUserID for template when status is userId updation.
When the account status is userIdUpdate ,then get the user from the
newly added method in base user directory.
